### ISSUE TITLE
[feature request] new search on top search button pressed 2x

### ISSUE THREAD

**Issue by nikmurder** [Franiac/TwitchLeecher#628]
ik it's minor af but it would b a lot easier if second press of the top search button would clear the previous search results and start a new search instead of doing nothing
ik there is a "new search" button in the lower part

also is there any chance we could play the vods from the search if we have streamlink installed?
if you want just part it's easier than going back and forth to browser finding the vod finding the spot going back etc

**Comment by Franiac** [comment]
The "Search" and "Download" buttons are for switching between those exact views, nothing more. Double-assigning buttons is the worst thing one could do in a software, that's a big nono. That's why we separate things between navigation events and actions.

"Search" means "show me the search panel and if I am already on it, don't do anything", and "New Search" is an action that starts a new search. I don't understand why this should be changed.

For the second part: I don't know Steamlink but it seems to be something like a Chromecast for mobile devices? What does that have to do with playing back videos in TL? I don't really understand what you mean there.

**Comment by nikmurder** [comment]
> I don't know Steamlink but it seems to be something like a Chromecast for mobile devices? What does that have to do with playing back videos in TL? I don't really understand what you mean there.

stReamlink (https://streamlink.github.io/) is to watch twitch in vlc (or other player) and not in browser
so basically to be able to watch the vod online to e.g. find the timing of the part you want b4 downloading it
all within TL and not use the twitch search in browser and their builtin player and then go back to TL and input the numbers
just like a play button (or click) on the vods u already have in ur search results in TL would b a lot faster

thank you for your answers and time and great software!

**Comment by Franiac** [comment]
Aha, the "R" makes a big difference. But anyway... I still don't really get what you are looking for. You want me to make TL Streamlink compatible? So basically turn TL into a video player with the ability to choose a part of that given video?

**Comment by nikmurder** [comment]
no, i meant make a "play" button next to "download" that would stream the vod in vlc or if that is not an option - even in browser player (which again is horrible imo)
the idea is to be able to choose the timings of the vod to download without going back to browser, searching for the vod (w/ that horrible twitch search that is barely usable), finally finding the time u want then going back into TL input them there and finally download
so basically avoid the twitch website functionality as much as possible in favor of TL + VLC + streamlink

**Comment by Franiac** [comment]
I don't get it... you can just click a video in the search view and the browser opens up the VOD. You don't need to find the video by yourself on the Twitch website.

Furthermore in the preferences you already can choose to open the VOD in VLC or any other video player.

Implementing a full blown streaming-tool just to be able to somehow select the timestamps is honestly overkill.

**Comment by nikmurder** [comment]
> I don't get it... you can just click a video in the search view and the browser opens up the VOD. You don't need to find the video by yourself on the Twitch website.
oh i did not know that! thanx!
and thank you again for the great software and your time answering!
